packages/components/src/detect-focus-outside/README.md

```diff
@@ -0,0 +1,36 @@
+# DetectFocusOutside
+
+`DetectFocusOutside` is a component used to help detect when focus leaves an element.
+
+This can be useful to disambiguate focus transitions within the same element. A React `blur` event will fire even when focus transitions to another element in the same context. `DetectFocusOutside` will only invoke its callback prop when focus has truly left the element.
```

Comment:
How about using _component tree_ instead of _element_ in the wording? It reflects better the use case we try to cover here. Example:

> `DetectFocusOutside` is a component used to help detect when focus leaves a component tree.
>
> This is useful to detect whether focus has transitioned to another element outside a particular component tree, in which case `DetectFocusOutside` will invoke the `onFocusOutside` callback.

Reply:
Yeah, I agree that's an improvement. "Leaves an element" could definitely be misinterpreted as the same as the original problem of "focus transitions from one element to another within the same component tree".

Reply:
Updated in 65c6232.